File: server/src/routes/catalog.ts
import { Router } from "express";
import { getPatterns, ATTACK_PATTERNS } from "../catalog/index.js";

export const catalogRouter = Router();

catalogRouter.get("/patterns", (req, res) => {
  const limit = req.query.limit ? Number(req.query.limit) : undefined;
  const category = req.query.category as string | undefined;
  let patterns = getPatterns(limit);
  if (category) {
    patterns = patterns.filter((p) => p.category === category);
  }
  res.json({
    total: ATTACK_PATTERNS.length,
    count: patterns.length,
    patterns,
  });
});

File: server/app/routes/catalog.py
from fastapi import APIRouter, Query

from app.catalog import get_patterns, load_patterns

router = APIRouter(prefix="/api/catalog", tags=["catalog"])


@router.get("/patterns")
def patterns(limit: int = Query(100, ge=1, le=500)):
    all_patterns = load_patterns()
    return {"total": len(all_patterns), "patterns": get_patterns(limit)}
